src/ufuncs.c

```diff
@@ -502,24 +485,36 @@ static void Y_i_func(char **args, npy_intp *dimensions,
 {
     FuncGEOS_Y_i *func = (FuncGEOS_Y_i *)data;
     void *context_handle = geos_context[0];
-    int ret;
     GEOSGeometry *in1;
 
     UNARY_LOOP {
-        INPUT_Y;
-        ret = func(context_handle, in1);
-        if ((ret < 0) | (ret > NPY_MAX_INT)) {
-            RAISE_ILLEGAL_GEOS;
-            return;
+        /* get the geometry: return on error */
+        if (!get_geom(*(GeometryObject **)ip1, &in1)) { return; }
+        if (in1 == NULL) {
+            /* NaG results in -1 */
+            *(npy_int *)op1 = -1;
+        } else {
+            /* trust that GEOS calls the error handler if necessary */
```

Comment:
This is mentioned many times, but wondering, do we actually test this? (in which occasion would GEOS raise an error? sometimes one invalid geometries?)

Reply:
There are some cases tested:
https://github.com/caspervdw/pygeos/search?q=pygeos.GEOSException&unscoped_q=pygeos.GEOSException

GEOS should do a callback to the error message handler (`HandleGEOSError`), which I defined such that a GEOSException is set. I have never encountered the situation where GEOS didn't do this.